Code_Exercises/OneMKL_gemm/solution_onemkl_buffer_gemm.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ SYCL Academy (c)
+
+ SYCL Academy is licensed under a Creative Commons
+ Attribution-ShareAlike 4.0 International License.
+
+ You should have received a copy of the license along with this
+ work.  If not, see <http://creativecommons.org/licenses/by-sa/4.0/>.
+
+ SYCL Quick Reference
+ ~~~~~~~~~~~~~~~~~~~~
+
+ // oneMKL APIs: https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/gemm.html#onemkl-blas-gemm
+
+ // DGEMM: https://www.intel.com/content/www/us/en/docs/onemkl/tutorial-c/2021-4/multiplying-matrices-using-dgemm.html
+
+*/
+
+#include <random>
+#include <iostream>
+#include <limits>
+
+#include <sycl/sycl.hpp>
+#include <oneapi/mkl/blas.hpp>
+
+
+// Matrix size constants
+#define SIZE 4800 // Must be a multiple of 8.
+#define M SIZE / 8
+#define N SIZE / 4
+#define P SIZE / 2
+
+using T = double;
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+
+bool ValueSame(T a, T b) { return std::fabs(a - b) < 1.0e-08; }
+int VerifyResult(sycl::host_accessor<T,1> & c_A, T *c_B) {
+  bool MismatchFound = false;
+
+  for (size_t i = 0; i < M; i++) {
+    for (size_t j = 0; j < P; j++) {
+      if (!ValueSame(c_A[i * P + j], c_B[i * P + j])) {
+        std::cout << "fail - The result is incorrect for element: [" << i << ", " << j
+                  << "], expected: " << c_A[i * P + j] << " , but got: " << c_B[i * P + j]
+                  << std::endl;
+        MismatchFound = true;
+      }
+    }
+  }
+
+  if (!MismatchFound) {
+    std::cout << "SUCCESS - The results are correct!" << std::endl;
+    return 0;
+  } else {
+    std::cout << "FAIL - The results mis-match!" << std::endl;
+    return -1;
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+
+void print_device_info(sycl::queue& Q) {
+    std::string sycl_dev_name, sycl_runtime, sycl_driver;
+    sycl_dev_name = Q.get_device().get_info<sycl::info::device::name>();
+    sycl_driver = Q.get_device().get_info<sycl::info::device::driver_version>();
+    sycl_runtime = Q.get_device().get_info<sycl::info::device::version>();
+    std::cout << "Running on " << sycl_dev_name.c_str() << ", SYCL runtime: v" << sycl_runtime.c_str() << ", driver version: " << sycl_driver.c_str() << std::endl;
```

Comment:
Nit: there are a few long lines like this one. Could you run clang-format on all of the cpp files for a consistent style?

Reply:
(Any clang-format style should be OK)

Reply:
Looks like you've done this. Thanks!